src/main/java/net/mcreator/workspace/references/ReferencesFinder.java

```diff
@@ -46,134 +46,111 @@
 
 public class ReferencesFinder {
 
-	public static Set<ModElement> searchModElementUsages(Workspace workspace, ModElement element) {
-		Set<ModElement> elements = new HashSet<>();
-
-		String query = element.getName();
-		workspace.getModElements().stream().filter(me -> !me.equals(element)).forEach(me -> {
-			GeneratableElement ge = me.getGeneratableElement();
-			if (anyValueMatches(ge, String.class, e -> e.isAnnotationPresent(ModElementReference.class), (a, t) -> {
-				ModElementReference ref = a.getAnnotation(ModElementReference.class);
-				return !Set.of(ref.defaultValues()).contains(t) && query.equals(
-						GeneratorWrapper.getElementPlainName(t));
-			})) {
-				elements.add(me);
-			} else if (anyValueMatches(ge, MappableElement.class, e -> e.isAnnotationPresent(ModElementReference.class),
-					(a, t) -> query.equals(GeneratorWrapper.getElementPlainName(t.getUnmappedValue())))) {
-				elements.add(me);
-			} else if (anyValueMatches(ge, Procedure.class, e -> e.isAnnotationPresent(ModElementReference.class),
-					(a, t) -> t.getName() != null && !t.getName().isEmpty() && !t.getName().equals("null")
-							&& query.equals(t.getName()))) {
-				elements.add(me);
-			} else if (anyValueMatches(ge, String.class, e -> e.isAnnotationPresent(BlocklyXML.class),
-					(a, t) -> t.contains(">CUSTOM:" + query + "</field>") || t.contains(">" + query + "</field>"))) {
-				elements.add(me);
-			}
-		});
+	/**
+	 * Convenience method to provide a list of all generatable elements contained in the provided workspace in a thread-safe manner.
+	 * List returned by this method can be used in parallel streams as long as all other methods in the said parallel stream are
+	 * also thread safe. Thus, e.g. calling ModElementManager within the parallel stream later would still cause problems.
+	 *
+	 * @param workspace Workspace to obtain generatable elements from.
+	 * @return List of all generatable elements contained in the provided workspace.
+	 */
+	private static List<GeneratableElement> getGeneratableElements(Workspace workspace) {
+		return workspace.getModElements().stream().map(ModElement::getGeneratableElement).collect(Collectors.toList());
+	}
 
-		return elements;
+	public static Set<ModElement> searchModElementUsages(Workspace workspace, ModElement element) {
+		//@formatter:off

```

Comment:
What's the reason for using this? Is the code unreadable otherwise?

Reply:
Yes

Reply:
So is it just me that I don't understand how this is more readable? #nooffense

Reply:
Ok, so this (formatted)

![image](https://github.com/MCreator/MCreator/assets/16374228/e0846dd5-435e-4de7-b997-abb4a04c7e49)

is more readable than this (formatted off):

![image](https://github.com/MCreator/MCreator/assets/16374228/18040bbd-2b11-430b-aa85-41ca91d80db5)
 
?

Reply:
I clearly am not robot enough to think first one is more readable

Reply:
Oh, so your formatter rebuilds the entire thing and it looks that weird after? Sorry, _I did not know that._

Reply:
> your formatter

Well I don't remember writing formatter of IDEA, but yes, it packs things together and in those cases, we use this flag to stop it

Reply:
That's sad (for IDEA), but if this format is a solution, I'll (have to) get used to it.